### PR TITLE
fix(project-access): add error code for cds mismatch

### PR DESCRIPTION
**Pull request by Klaus-Keller**
Add `error.code` with value `CDS_VERSION_MISMATCH` to error that is thrown, if a fallback to global `@sap/cds` results in a different major version then the project has. See also https://github.com/SAP/open-ux-tools/pull/1823.

### CODE REVIEW

**Comment by IainSAP** [comment]
@Klaus-Keller Can we also add an error code for the error :
```
[Error: Error while reading CAP service metadata. Path: '/Users/i057546/git/open-ux-tools/packages/odata-service-inquirer/test/unit/prompts/cap-project/fixtures/bookshop', service uri: 'odata/v4/incident/', error: 'Error: Could not load cds module. Attempt to load module @sap/cds from project threw error 'Error: Module '@sap/cds' not installed in project '/Users/i057546/git/open-ux-tools/packages/odata-service-inquirer/test/unit/prompts/cap-project/fixtures/bookshop'.
    Error: Path to module not found.', attempt to load module @sap/cds from @sap/cds-dk threw error 'Error: Module path not found''}]
```

Where an install has not bee done on the project?

**Review by IainSAP — APPROVED** [review]
Thanks @Klaus-Keller 

- Approving as needed by new module `@sap-ux/odata-servivce-inquirer`
- Further error codes can be added as required
- Test as expected

File: packages/project-access/src/project/cap.ts [hunk 1]
```diff
@@ -408,9 +408,11 @@ async function loadCdsModuleFromProject(capProjectPath: string, strict: boolean
         if (typeof cdsDependencyVersion === 'string') {
             const globalCdsVersion = cds.version;
             if (getMajorVersion(cdsDependencyVersion) !== getMajorVersion(globalCdsVersion)) {
-                throw Error(
+                const error = new Error(
                     `The @sap/cds major version (${cdsDependencyVersion}) specified in your CAP project is different to the @sap/cds version you have installed globally (${globalCdsVersion}). Please run 'npm install' on your CAP project to ensure that the correct CDS version is loaded.`
-                );
+                ) as Error & { code: string };
+                error.code = 'CDS_VERSION_MISMATCH';
```
Comment (IainSAP) [inline packages/project-access/src/project/cap.ts:414]: Should these error code strings be exported as consts for import and re-use in other modules?

Comment (Klaus-Keller) [reply packages/project-access/src/project/cap.ts:414]: We could add the error codes to JSDoc.

**Review by zdravko-georgiev — APPROVED** [review]
Code straightforward and easy to understand
Excellent test coverage
Did NOT tested manually

**Review by kranthie-sap — APPROVED** [review]
* Verified that changeset config exists
* Changes to `cap.ts` and corresponding unit test look good. It is odd that the `Error` type used doesn't have the `code` property exposed. The workaround to augment the type looks good. More information about the issue has been discussed along with various solutions are in this thread - https://stackoverflow.com/questions/51523509/in-typescript-how-do-you-make-a-distinction-between-node-and-vanilla-javascript

**Pull request merged**